zio-http/src/main/scala/zio/http/HExit.scala

```diff
@@ -1,7 +1,7 @@
 package zio.http
 
 import zio.http.HExit.Effect
-import zio.{Cause, Trace, ZIO}
+import zio.{Cause, Tag, Trace, ZEnvironment, ZIO, ZLayer}
```

Comment:
Are we going to be able to replace this in favor of `zio.Exit` with appropriate optimizations applied?

Reply:
I opened a ticket about it few days ago. Should we do it in this PR now that 2.0.6 is out?

Reply:
(I think we would be able to simply replace it with `Exit` now)

Reply:
Pushed it here, with some workaround to override `flatMap/map/mapBoth` (also opened https://github.com/zio/zio/pull/7714 but it creates stack safety issues in streams)